os/kernel/task/task_prctl.c

```diff
@@ -200,16 +206,26 @@ int prctl(int option, ...)
 		} else {
 			return MSG_READ_YET;
 		}
+#else
+		sdbg("Not supported.\n");
+		err = ENOSYS;
+		goto errout;
+#endif
 	}
 	break;
 	case PR_MSG_REMOVE:
 	{
+#ifdef CONFIG_MESSAGING_IPC
 		int ret;
 		char *port_name = va_arg(ap, char *);
 		ret = messaging_remove_list(port_name);
 		return ret;
-	}
+#else
+		sdbg("Not supported.\n");
+		err = ENOSYS;
+		goto errout;
 #endif
```

Comment:
How about below to avoid duplicated codes?
```
#ifdef CONFIG_MESSAGING_IPC
case PR_MSG_READ:
  ....
  break;
case PR_xx:
  break;
#else
 case AA
 case BB
 case CC
   sdbg("Not supported cmd %d.\n", option);
   err = ;
   goto ;
#endif
```